native/acter/src/api/reactions.rs

```diff
@@ -0,0 +1,218 @@
+use acter_core::models::{self, ActerModel, AnyActerModel};
+use anyhow::{bail, Context, Result};
+use core::time::Duration;
+use futures::stream::StreamExt;
+use matrix_sdk::room::Room;
+use ruma_common::{EventId, OwnedEventId, OwnedTransactionId, OwnedUserId};
+use ruma_events::{reaction::ReactionEventContent, relation::Annotation, MessageLikeEventType};
+use std::ops::Deref;
+use tokio::sync::broadcast::Receiver;
+use tokio_stream::{wrappers::BroadcastStream, Stream};
+use tracing::trace;
+
+use super::{client::Client, RUNTIME};
+
+impl Client {
+    pub async fn wait_for_reaction(
+        &self,
+        key: String,
+        timeout: Option<Box<Duration>>,
+    ) -> Result<Reaction> {
+        let client = self.clone();
+        RUNTIME
+            .spawn(async move {
+                let AnyActerModel::Reaction(reaction) = client.wait_for(key.clone(), timeout).await? else {
+                    bail!("{key} is not a reaction");
+                };
+                let room = client
+                    .core
+                    .client()
+                    .get_room(&reaction.meta.room_id)
+                    .context("Room not found")?;
+                Ok(Reaction {
+                    client: client.clone(),
+                    room,
+                    inner: reaction,
+                })
+            })
+            .await?
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Reaction {
+    client: Client,
+    room: Room,
+    inner: models::Reaction,
+}
+
+impl Deref for Reaction {
+    type Target = models::Reaction;
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl Reaction {
+    pub fn event_id_str(&self) -> String {
+        self.inner.event_id().to_string()
+    }
+
+    pub fn sender(&self) -> OwnedUserId {
+        self.inner.meta.sender.clone()
+    }
+
+    pub fn origin_server_ts(&self) -> u64 {
+        self.inner.meta.origin_server_ts.get().into()
+    }
+
+    pub fn relates_to(&self) -> String {
+        self.inner.relates_to.event_id.to_string()
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct ReactionManager {
+    client: Client,
+    room: Room,
+    inner: models::ReactionManager,
```

Comment:
this should store the `event_id`, too.

Reply:
`event_id` can be known by `inner.event_id()`.